go-controller/pkg/ovn/ovn.go

```diff
@@ -112,20 +134,186 @@ func NewOvnController(kubeClient kubernetes.Interface, wf *factory.WatchFactory)
 		gatewayCache:             make(map[string]string),
 		loadbalancerClusterCache: make(map[string]string),
 		loadbalancerGWCache:      make(map[string]string),
+		serviceVIPToName:         make(map[ServiceVIPKey]types.NamespacedName),
+		serviceVIPToNameLock:     sync.Mutex{},
 	}
 }
 
-// Run starts the actual watching.
-func (oc *Controller) Run() error {
-	for _, f := range []func() error{oc.WatchNodes, oc.WatchPods, oc.WatchServices, oc.WatchEndpoints,
-		oc.WatchNamespaces, oc.WatchNetworkPolicy} {
+// Run starts the actual watching. Also initializes any local structures needed.
+func (oc *Controller) Run(stopChan chan struct{}) error {
+	_, _, err := util.RunOVNNbctl("--columns=_uuid", "list",
+		"port_group")
+	if err == nil {
+		oc.portGroupSupport = true
+	}
+
```

Comment:
@dcbw @putnopvut please remove lines L144-148, since they are already taken care of in oc.StartClusterMaster()

Reply:
@girishmg @markmc I fixed that up and force-pushed to this PR